packages/salesforcedx-vscode-automation-tests/test/specs/apexLsp.e2e.ts

```diff
@@ -0,0 +1,268 @@
+  // Allow time for LSP to process text input and prepare autocompletion suggestions
+  await pause(Duration.seconds(2));
+
+  const autocompletionOptions = await workbench.findElements(By.css('div.monaco-list-row.show-file-icons'));
+  const ariaLabel = await autocompletionOptions[0].getAttribute('aria-label');
+  expect(ariaLabel).to.contain('SayHello(name)');
+
+  await autocompletionOptions[0].click();
+  await textEditor.typeText("'Jack");
+  await textEditor.typeTextAt(7, 38, ';');
+  await textEditor.save();
+
+  // Allow time for LSP to process the changes and update the editor
+  await pause(Duration.seconds(2));
+  const line7Text = await textEditor.getTextAtLine(7);
+  expect(line7Text).to.include("ExampleClass.SayHello('Jack');");
+};
+
+const testLspRestart = async (testSetup: TestSetup, cleanDb: boolean): Promise<void> => {
+  const action = cleanDb ? 'with cleaned db' : 'alone';
+  log(`${testSetup.testSuiteSuffixName} - Cmd Palette: LSP Restart ${action}`);
+
+  if (cleanDb) {
+    const releaseDir = findReleaseDir();
+    const standardApexLibraryPath = path.join(PATHS.tools, releaseDir, 'StandardApexLibrary');
+    await removeFolder(standardApexLibraryPath);
+    expect(await getFolderName(standardApexLibraryPath)).to.equal(null);
+  }
+
+  const restartCommand = await executeQuickPick('Restart Apex Language Server');
+  const quickPicks = await restartCommand.getQuickPicks();
+  for (const quickPick of quickPicks) {
+    const label = await quickPick.getLabel();
+    if (label === (cleanDb ? 'Clean Apex DB and Restart' : 'Restart Only')) {
+      await quickPick.select();
+      break;
+    }
+  }
+  await verifyLspRestart(cleanDb);
+
+  if (cleanDb) {
+    const releaseDir = findReleaseDir();
+    expect(await getFolderName(path.join(PATHS.tools, releaseDir, 'StandardApexLibrary'))).to.equal(
+      'StandardApexLibrary'
+    );
+  }
+};
+
+const testStatusBarRestart = async (testSetup: TestSetup, cleanDb: boolean): Promise<void> => {
+  const action = cleanDb ? 'with cleaned db' : 'alone';
+  log(`${testSetup.testSuiteSuffixName} - Apex Status Bar: LSP Restart ${action}`);
+
+  const statusBar = await getStatusBarItemWhichIncludes('Editor Language Status');
+  await statusBar.click();
+
+  // Allow time for status bar menu to appear and be clickable
+  await pause(Duration.seconds(3));
+  const restartButton = getWorkbench().findElement(By.linkText('Restart Apex Language Server'));
+  await restartButton.click();
+
+  // Allow time for restart process to begin
+  const dropdown = await new InputBox().wait();
+  await selectQuickPickItem(dropdown, cleanDb ? 'Clean Apex DB and Restart' : 'Restart Only');
+  await verifyLspRestart(cleanDb);
+
+  if (cleanDb) {
+    const releaseDir = findReleaseDir();
+    expect(await getFolderName(path.join(PATHS.tools, releaseDir, 'StandardApexLibrary'))).to.equal(
+      'StandardApexLibrary'
+    );
+  }
+};
+
+describe('Apex LSP', async () => {
+  let testSetup: TestSetup;
```

Comment:
see https://github.com/lo1tuma/eslint-plugin-mocha/blob/main/docs/rules/no-async-suite.md.  I'm pretty sure it's not supposed to be async